libs/wire-subsystems/src/Wire/EnterpriseLoginSubsystem/Interpreter.hs

```diff
@@ -284,14 +285,15 @@ unauthorizeImpl ::
   Sem r ()
 unauthorizeImpl domain = do
   old <- lookupOrThrow domain
-  let new = old {domainRedirect = None} :: DomainRegistration
-  case old.domainRedirect of
-    PreAuthorized -> audit old new *> upsert new
-    Backend _ -> audit old new *> upsert new
-    NoRegistration -> audit old new *> upsert new
-    None -> pure ()
-    Locked -> throw EnterpriseLoginSubsystemOperationForbidden
-    SSO _ -> throw EnterpriseLoginSubsystemOperationForbidden
+  let new = old {settings = Nothing} :: DomainRegistration
+  case old.settings of
+    Nothing -> pure ()
+    Just DomainLocked -> throw EnterpriseLoginSubsystemOperationForbidden
+    Just DomainPreAuthorized -> audit old new *> upsert new
+    Just (DomainNoRegistration _) -> audit old new *> upsert new
+    Just (DomainForBackend _) -> audit old new *> upsert new
+    Just (DomainTeamInviteRestricted _) -> audit old new *> upsert new
```

Comment:
```suggestion
    Just (DomainTeamInviteRestricted _) -> pure ()
```

`DomainTeamInviteRestricted _` implies `domainRedirect == None`, and the old code says `None -> pure ()`.